Jenkinsfile

```diff
@@ -0,0 +1,174 @@
+    stage "packaging - docker prep"
+
+    // Docker environment to build packagings
+    dir('packaging-docker') {
+        git branch: packagingBranch, url: 'https://github.com/jenkinsci/packaging.git'
+        sh 'docker build -t jenkins-packaging-builder:0.1 docker'
+    }
+
+    stage "packaging - actually packaging"
+    // Working packaging code, separate branch with fixes
+    dir('packaging') {
+        deleteDir()
+
+        docker.image("jenkins-packaging-builder:0.1").inside("-u root") {
+            git branch: packagingBranch, url: 'https://github.com/jenkinsci/packaging.git'
+
+            try {
+                // Saw issues with unstashing inside a container, and not sure copy artifact plugin would work here.
+                // So, simple wget.
+                sh "wget -q ${currentBuild.absoluteUrl}/artifact/war/target/jenkins.war"
+                sh "make clean deb rpm suse BRAND=./branding/jenkins.mk BUILDENV=./env/test.mk CREDENTIAL=./credentials/test.mk WAR=jenkins.war"
+            } catch (Exception e) {
+                error "Packaging failed: ${e}"
+            } finally {
+                // Needed to make sure the output of the build can be deleted by later runs.
+                // Hackish, yes, but rpm builds as a numeric UID only user fail, so...
+                sh "chmod -R a+w target || true"
+                sh "chmod a+w jenkins.war || true"
+            }
+            dir("target/debian") {
+                def debFilesFound = findFiles(glob: "*.deb")
+                if (debFilesFound.size() > 0) {
+                    debFileName = debFilesFound[0]?.name
+                }
+            }
+
+            dir("target/rpm") {
+                def rpmFilesFound = findFiles(glob: "*.rpm")
+                if (rpmFilesFound.size() > 0) {
+                    rpmFileName = rpmFilesFound[0]?.name
+                }
+            }
+
+            dir("target/suse") {
+                def suseFilesFound = findFiles(glob: "*.rpm")
+                if (suseFilesFound.size() > 0) {
+                    suseFileName = suseFilesFound[0]?.name
+                }
+            }
+            step([$class: 'ArtifactArchiver', artifacts: 'target/**/*', fingerprint: true])
+        }
+
+    }
+
+}
+
+stage "Package testing"
+
+if (runTests) {
+    // NOTE: As of now, a lot of package tests will fail. See https://issues.jenkins-ci.org/issues/?filter=15257 for
+    // possible open JIRAs.
+
+    // Basic parameters
+    String artifactName = (binding.hasVariable('artifactName')) ? artifactName : 'jenkins'
+    String jenkinsPort = (binding.hasVariable('jenkinsPort')) ? jenkinsPort : '8080'
+
+    // Set up
+    String debfile = "artifact://${env.JOB_NAME}/${env.BUILD_NUMBER}#target/debian/${debFileName}"
+    String rpmfile = "artifact://${env.JOB_NAME}/${env.BUILD_NUMBER}#target/rpm/${rpmFileName}"
+    String susefile = "artifact://${env.JOB_NAME}/${env.BUILD_NUMBER}#target/suse/${suseFileName}"
+
+    node("docker") {
```

Comment:
I didn't bother initially because this is so short - the real work is all done off in stuff from packaging.git, which I'll give another PR to add equivalent support for. But hey, I'll change this here. =)